### ISSUE TITLE
doReturnJSON responses are rendered with Content Type `text/plain`

### ISSUE THREAD

**Issue by mscheepmaker** [PawelAdamski/HttpClientMock#14]
Thanks for providing this, very helpful 👍

I ran into an issue when using `doReturnJson()`:  Content-Type `application/json` appears to be added as a response header instead of an entity header. A default entity header Content-Type `text/plain` is added by StringEntity, which can lead to problems when processing the response.

As a work-around we can use a custom Action response to create the StringEntity with the correct mime-type, so it's only a minor issue.

**Comment by PawelAdamski** [comment]
Hi @mscheepmaker 
To be honest I don't understand the problem. Currently `doReturnJson` method adds `application/json` content-type and in my opinion it is a correct type. If I understand your request correct you would like to have Content-Type `text/plain`. Is this correct?

Please put here your work-around so I could understand what you want to achieve.

**Comment by mscheepmaker** [comment]
Hi @PawelAdamski,
Sorry I wasn't clear enough. `application/json` is of course the expected content-type. The problem is that the response generated by the mockClient contains two Content-Type headers, one with value `application/json` (added by your code) and the other with `text/plain` (added by `StringEntity`). My code is using `org.apache.http.entity.ContentType.get(HttpEntity entity)` to read the Content-Type header from the response, which returns 'text/plain'.

The work-around:

```
mockClient.onGet("http://host/metadata").doAction(request -> {
            BasicHttpResponse response = new BasicHttpResponse(new ProtocolVersion("http", 1, 1), 200, "ok");
            response.setEntity(new StringEntity("{\"foo\":\"bar\"}", ContentType.APPLICATION_JSON));
            return response;
        });
```

**Comment by basalt79** [comment]
feel free to take https://github.com/PawelAdamski/HttpClientMock/pull/17 for this item.

**Comment by PawelAdamski** [comment]
Thanks @basalt79  for your contribution.
I've added three comments to your PR.

**Comment by PawelAdamski** [comment]
Fixed in version 1.6.0
